p2p/host/relay/autorelay.go

```diff
@@ -92,7 +92,9 @@ func (h *AutoRelayHost) background(ctx context.Context) {
 
 	for {
 		wait := autonat.AutoNATRefreshInterval
-		switch h.autonat.Status() {
+		status := h.autonat.Status()
+		h.Ids.SetNatStatus(status)
```

Comment:
that's gross -- we should embed AutoNAT in basic_host if identify needs it or pass it a reference.